### ISSUE TITLE
Ability to load compressed models

### ISSUE THREAD

**Issue by cjrh** [messense/fasttext-rs#11]
Hi! Thank you for this project.

I was curious whether you had any plans to implement support for compressed models?

- They are described in detail in this blog post: https://towardsdatascience.com/compressing-unsupervised-fasttext-models-eb212e9919ca
- Here are several of the models mentioned in the blog post: https://github.com/avidale/compress-fasttext/releases/tag/gensim-4-draft
- This is the Python library that provides support for loading compressed models: https://github.com/avidale/compress-fasttext

**Comment by messense** [comment]
fasttext-rs is a binding to the cpp fastText library, I think compressed models support is out of the scope.

But I'm happy to accept a PR that implements it under a feature gate.

**Comment by cjrh** [comment]
Thank you for your response :)

Yes, I understand. Since I created this issue, I learned a lot more about how the compressed models in that blog post actually work. They are no longer actual "fasttext" models after compression. Instead, they're special gensim "keyed vectors". This is why I've been unable to load them using different libraries, including finalfusion for example.

I think what we'd have to do to be able to use these compressed models is actually reimplement support for those "keyed vectors" files in rust, which I agree is completely out of scope for fasttext-rs.
